File: objective-c/vcdiff/Project/blocktext.m


#import "blocktext.h"

//definition of blocktext class

@implementation blocktext

//Function to split and add the source text into array of blocks according to the given blocksize
-(void)initwith:(NSMutableString *)orgtxt:(int) bsize
{
	NSMutableString *s1=[[NSMutableString alloc]init];
	blocks=[[NSMutableArray alloc] init];
	block *b;
	originaltext=orgtxt;
	blocksize=bsize;
	int i,len,endindex;//endindex used to obtain substring from originaltext
	len=[originaltext length];
	for(i=0;i<len;i+=blocksize)
	{
		
		endindex=i+blocksize>=len?len:i+blocksize;//endindex must be len for the last block
		s1=[NSMutableString stringWithFormat:@"%@",[originaltext substringWithRange:NSMakeRange(i,endindex-i)]];//Splits the string from i to endindex
		b=[[block alloc] init]; //Allocates memory for each blocks
		[b initwith:s1:i];
		[blocks addObject:b];//Adds to the array
	}
			
}
-(NSMutableArray *)getblocks
{
	
	return blocks;
}
-(NSMutableString *)getoriginaltext
{
	return originaltext;
}
-(int)getblocksize
{
	return blocksize;
}
@end

File: java-old/com.neoba.dsync.vcdiff/src/com/neoba/dsync/vcdiff/BlockText.java
package com.neoba.dsync.vcdiff;

import java.util.ArrayList;
import java.util.List;

public class BlockText {

    String originalText;
    int blockSize;
    List<Block> blocks;
    /**
     * creates a list of Blocks
     * @param originalText
     * @param blockSize 
     */
    BlockText(String originalText, int blockSize) {
        this.originalText = originalText;
        this.blockSize = blockSize;
        int i;
        int len = originalText.split("").length - 1;
        int endIndex;
        blocks = new ArrayList<>();

        for (i = 0; i < len; i += blockSize) {
            endIndex = i + blockSize >= len ? len : i + blockSize;
            blocks.add(new Block(originalText.substring(i, endIndex), i));
        }
    }

    List<Block> getBlocks() {
        return this.blocks;
    }

    String getOriginalText() {
        return this.originalText;
    }

    int getBlockSize() {
        return this.blockSize;
    }
}
